fix: Skip button reloading from cached images

ItsAbhinavM opened the pull request:
# Description
Fixes #319 
This PR fixes the problem with the skip button which does not loads from the backend but instead endlessly loop over the same 10 images which was cached until we accept/reject any image in that set.

The root cause was just a mismatch on the backend, since it was supposed to call `round_juror.flags['skip']` but it was trying to read from `round_juror.skip` column which does not exist.
Moreover the frontend didn't even had a POST method to convey the backend about the skip, due to which the backend never knew about the skip.

mahmoud reviewed (changes requested):
Interesting! I believe skip worked before the new frontend. I half expected this to be primarily frontend code changes, but I see it's only backend changes. Are there supposed to be frontend changes too?

Also, when updating the backend we should probably be adding tests for repro/regression protection, as well.

ItsAbhinavM commented on montage/rdb.py:
```diff
@@ -2559,33 +2560,24 @@ def get_tasks_by_id(self, task_ids):
+        )
+        
         round_juror = self._get_round_juror(round_id)
-        skip = round_juror.skip
```
Yes, this was the root cause! The code was trying to read `round_juror.skip` (which doesn't exist) instead of `round_juror.flags['skip']`. It didn't crash because it just returned None, so the skip filter was silently never applied. This caused skipped images to keep reappearing.

ItsAbhinavM commented on montage/rdb.py:
```diff
@@ -2824,19 +2816,35 @@ def edit_rating(self, task, value, review=''):
+            flag_modified(round_juror, 'flags')
+            
+            self.rdb_session.add(round_juror)
+            self.rdb_session.commit()
```
Sry about that, I'll remove that line

ItsAbhinavM commented:
> Interesting! I believe skip worked before the new frontend. I half expected this to be primarily frontend code changes, but I see it's only backend changes. Are there supposed to be frontend changes too?
> 
> Also, when updating the backend we should probably be adding tests for repro/regression protection, as well.

YES, the frontend changes are absolutely required. For some reason that I checked now, the changes I made on `vote/`  directory was not even staged for this PR. I added a seperate call for this skiptask since this was the one who called the whole issue. I added it on both `VoteRating.vue` and `VoteYesNo.vue`. I'll be adding these along with the changes you have requested. Now everything should go as planned for this bugfix :)

ItsAbhinavM commented:
I've also attached the recording of this fix on the issue. [Here](https://github.com/hatnote/montage/issues/319#issuecomment-3701785329)

mahmoud reviewed (changes requested):
That's more like it! Frontend + backend starting to make sense. Only further idea I have is to enhance the current skip test (which doesn't do any validation really) here: https://github.com/hatnote/montage/blob/ec3a1a795b422d78ee91332329a334de04683b37/montage/tests/test_web_basic.py#L440-L444

We just need to add a confirmation that the backend marked it as skipped, either through the return or another API call.

ItsAbhinavM commented:
> That's more like it! Frontend + backend starting to make sense. Only further idea I have is to enhance the current skip test (which doesn't do any validation really) here:
> 
> https://github.com/hatnote/montage/blob/ec3a1a795b422d78ee91332329a334de04683b37/montage/tests/test_web_basic.py#L440-L444
> 
> We just need to add a confirmation that the backend marked it as skipped, either through the return or another API call.

I've added that test there now, can you check and tell me whether any more changes is required!
Thanks!
